V1/Services/OrganisationService.cs

```diff
@@ -122,15 +128,24 @@ public async Task<Response> UpdateCountAsync(string id, UpdateCountRequest paylo
 
         public async Task<Response> MobileEntry(string organisationId, MobileEntryRequest payload)
         {
-            var wallet = await _walletService.CreateWalletAndSms(payload.WalletRequest);
+            var walletRequest = new CreateWalletRequest
+            {
+                MobileNumber = payload.MobileNumber
+            };
+            var wallet = await _walletService.CreateWallet(walletRequest);
+
+            var session = await _sessionService.CreateSession(payload.MobileNumber);
+
+            var organisation = await _organisationRepository.GetAsync(Guid.Parse(organisationId));
```

Comment:
You don't check if organisation is null. Just continuing and assuming it does exist.